queries/c_sharp/indents.scm

```diff
@@ -0,0 +1,117 @@
+[
+ (property_declaration)
+ ; (field_declaration)
+ (variable_declaration)
+ (accessor_declaration)
+
+ ; (declaration_list)
+
+ (parameter_list)
+ (type_parameter_list)
+ (accessor_list)
+ (attribute_argument_list)
+ (base_list)
+ (argument_list)
+
+ (binary_expression)
+ (expression_statement)
+ (return_statement)
+ (local_declaration_statement)
+ (local_function_statement)
+ (yield_statement)
+ (fixed_statement)
+ (unsafe_statement)
+ (labeled_statement)
+ (for_each_statement)
+ (for_statement)
+ (if_statement)
+ (switch_expression)
+ (switch_section)
+ (switch_statement)
+ (while_statement)
+ (lambda_expression)
+ (do_statement)
+ (checked_statement)
+ (try_statement)
+ (using_statement)
+ (local_function_statement)
+ (object_creation_expression)
+ (array_creation_expression)
+ (implicit_object_creation_expression)
+
+ ] @indent.begin
+
+(method_declaration body: _ @indent.begin)
+(namespace_declaration body: _ @indent.begin)
+(record_declaration body: _ @indent.begin)
+(class_declaration body: _ @indent.begin)
+(struct_declaration body: _ @indent.begin)
+(constructor_declaration body: _ @indent.begin)
+(interface_declaration body: _ @indent.begin)
+(enum_declaration body: _ @indent.begin)
+
+(property_declaration (accessor_list "{" @indent.branch) @indent.dedent)
+
+(switch_section
+  (case_switch_label) (case_switch_label) @indent.branch)
+
+
+(if_statement (block "{" @indent.branch))
+(if_statement alternative: (if_statement) @indent.dedent)
```

Comment:
The if_statement should be similar enough to C that I suggest you can use the C's indent queries version, with nodes modification (i.e. `(compound_statement) -> (block))

Reply:
I remember I was looking onto C's version, but I understood only like half of it, so i did this, which I undestand. I think there is subtle difference in tree for C and C#. I generally had hard time to debug the thing so I ended up with kind of first working solution.